toolkit/tools/srpmdownloader/srpmdownloader.go

```diff
@@ -0,0 +1,232 @@
+	}
+
+	// A pack list may be provided, if so only pack this subset.
+	// If none is provided, pack all srpms.
+	srpmList, err := srpm.ParsePackListFile(*srpmListFile)
+	logger.PanicOnError(err)
+
+	logger.Log.Infof("SRPM list %s", srpmList)
+
+	err = getSRPMQueryWrapper(*specsDir, *distTag, *buildDir, *outDir, *workerTar, *workers, *runCheck, srpmList, templateSrcConfig)
+	logger.PanicOnError(err)
+
+}
+
+// getSRPMQueryWrapper wraps getSRPMQuery to conditionally run it inside a chroot.
+// If workerTar is non-empty, packing will occur inside a chroot, otherwise it will run on the host system.
+func getSRPMQueryWrapper(specsDir, distTag, buildDir, outDir, workerTar string, workers int, runCheck bool, srpmList []string, templateSrcConfig sourceRetrievalConfiguration) (err error) {
+	var chroot *safechroot.Chroot
+	originalOutDir := outDir
+	if workerTar != "" {
+		const leaveFilesOnDisk = false
+		chroot, buildDir, outDir, specsDir, err = srpm.CreateSRPMChroot(workerTar, buildDir, outDir, specsDir)
+		if err != nil {
+			return
+		}
+		defer chroot.Close(leaveFilesOnDisk)
+	}
+
+	doCreateAll := func() error {
+		err = getSRPMQuery(specsDir, distTag, buildDir, outDir, workers, runCheck, srpmList, templateSrcConfig)
+		return err
+	}
+
+	if chroot != nil {
+		logger.Log.Info("Grabbing SRPMs URL inside a chroot environment")
+		err = chroot.Run(doCreateAll)
+	} else {
+		logger.Log.Info("Grabbing SRPMs URL SRPMs in the host environment")
+		err = doCreateAll()
+	}
+
+	if err != nil {
+		return err
+	}
+
+	// If this is container build then the bind mounts will not have been created.
+	// Copy the chroot output to host output folder.
+	if !buildpipeline.IsRegularBuild() {
+		srpmsInChroot := filepath.Join(chroot.RootDir(), outDir)
+		err = directory.CopyContents(srpmsInChroot, originalOutDir)
+		if err != nil {
+			return err
+		}
+	}
+
+	return
+}
+
+// getSRPMQuery queries for the name, version and release of the SRPM
+func getSRPMQuery(specsDir, distTag, buildDir, outDir string, workers int, runCheck bool, srpmList []string, srcConfig sourceRetrievalConfiguration) (err error) {
+	const (
+		emptyQueryFormat = ``
+		querySrpm        = `%{NAME}-%{VERSION}-%{RELEASE}.src.rpm`
+	)
+	// Find the general SPEC query arguments
+	defines := rpm.DefaultDefines(runCheck)
+	defines[rpm.DistTagDefine] = distTag
+	arch, err := rpm.GetRpmArch(runtime.GOARCH)
+	if err != nil {
+		logger.Log.Warn(err)
+		return
+	}
+	specFiles, err := srpm.FindSPECFiles(specsDir, srpmList)
+	if err != nil {
+		return
+	}
+
+	// Parse each SPEC for name and version and hydrate SRPM
+	for _, specfile := range specFiles {
+		sourcedir := filepath.Dir(specfile)
+		var packageSRPMs []string
+		packageSRPMs, err = rpm.QuerySPEC(specfile, sourcedir, querySrpm, arch, defines, rpm.QueryHeaderArgument)
+		if err != nil {
+			logger.Log.Warn(err)
+			return err
+		}
+		packageSRPM := packageSRPMs[0]
+		downloadSRPM(packageSRPM, outDir, srcConfig)
+	}
+	return err
+}
```

Comment:
[Ignorable] Minor clean-up:
```suggestion
	const (
		emptyQueryFormat = ``
		querySrpm        = `%{NAME}-%{VERSION}-%{RELEASE}.src.rpm`
	)
	// Find the general SPEC query arguments
	defines := rpm.DefaultDefines(runCheck)
	defines[rpm.DistTagDefine] = distTag

	arch, err := rpm.GetRpmArch(runtime.GOARCH)
	if err != nil {
		logger.Log.Warn(err)
		return
	}

	specFiles, err := srpm.FindSPECFiles(specsDir, srpmList)
	if err != nil {
		return
	}

	// Parse each SPEC for name and version and hydrate SRPM
	for _, specfile := range specFiles {
		var packageSRPMs []string
		sourcedir := filepath.Dir(specfile)
		packageSRPMs, err = rpm.QuerySPEC(specfile, sourcedir, querySrpm, arch, defines, rpm.QueryHeaderArgument)
		if err != nil {
			logger.Log.Warn(err)
			return err
		}

		downloadSRPM(packageSRPMs[0], outDir, srcConfig)
	}

	return
}
```